### PR TITLE
Cart crud

### PR DESCRIPTION
**Pull request by corneliusellen**
# Added Features:
* Added Increase Quantity, Decrease Quantity, Remove and index functionality for accessories in cart.
* Added a "Total Value" method to cart model to show subtotal of cart.

# Benefits (why)
* Instead of "Undo" in the flash message leading back to the accessory show page, I had it undo the delete function and add that method back to the cart. I think this is more like actual e-commerce sites.

# Possible Drawbacks (optional)
* The destroy action for carts is pretty long and could be refactored. The challenge is that there are two paths a user can delete an accessory - by decreasing the quantity and removing the accessory (not matter the quantity) completely.  I'm not sure how to start refactoring this @icorson3 @AliSchlereth 

# Checklist:

Answer the following questions -
* Are all tests passing? Y
* Have you tested your new feature? Y
* Have you tested it on localhost? Y

# Additional Comments:
* In order to add links/buttons in flash messages, you must add ".html_safe" to the end of the flash message string in the controller AND in the application.html view.

### CODE REVIEW

**Review by lnchambers — APPROVED** [review]

**Comment by AliSchlereth** [comment]
@corneliusellen Will you tag me in the specific line/method you want me to look at?

**Comment by AliSchlereth** [comment]
@corneliusellen Reading your PR comment - without reading through the code - make sure you're matching what the spec asks for. I'd implement a link back to the accessory show, and ask your Project Manager (me) if it is acceptable to add the functionality of the undo button. When asking to change something, make sure to include user story you'd like to alter. Remember the spec represents exactly what your client has asked for.

**Comment by corneliusellen** [comment]
@AliSchlereth Can I add functionality to the Undo button that instead actually undos the action the user just did (which was delete an item from their cart)?

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+      @cart = Cart.new(session[:cart])
+      @cart.decrease_quantity(params[:id])
+      session[:cart] = @cart.contents
+      redirect_to carts_path
+    else
+      @accessory = Accessory.find(params[:id])
+      @cart = Cart.new(session[:cart])
+      @cart.remove_accessory(params[:id])
+      session[:cart] = @cart.contents
+      flash[:notice] = %Q(Successfully removed #{@accessory.name} from your cart. #{view_context.button_to("Undo", carts_path(accessory: @accessory.id, previous_page: "cart_page"))}).html_safe
```
[the remark this answers is not in the document]
Comment (corneliusellen) [reply app/controllers/carts_controller.rb]: Definitely!

**Comment by AliSchlereth** [comment]
@corneliusellen I promise I'm not trying to be difficult for the sake of being difficult. 
You need to include the complete user story you're requesting to change though before, as PM I'm willing to consider your request.

**Comment by corneliusellen** [comment]
I totally forgot to provide the user story:
User Adds item to cart.
User clicks on cart.
User sees item in cart. 
User removes item from cart and clicks the link"Undo" on a flash message.
User is on the cart page and sees the original item back in her cart.

**Comment by AliSchlereth** [comment]
@corneliusellen This sounds good - as long as the tittle of the item is also a link back to the item show.

**Pull request merged**